src/hotspot/cpu/aarch64/c1_Runtime1_aarch64.cpp

```diff
@@ -602,6 +604,7 @@ OopMapSet* Runtime1::generate_code_for(StubID id, StubAssembler* sasm) {
   // for better readability
   const bool must_gc_arguments = true;
   const bool dont_gc_arguments = false;
+  const bool does_not_return = true;
```

Comment:
I wonder if an `enum` would read better here.

Something like

```
enum may_return_t {
  does_not_return, may_return
};

class StubFrame {

  StubFrame(StubAssembler* sasm, const char* name, bool must_gc_arguments, may_return_t may_return);

};
```

Reply:
Ok. I was keeping to the existing style of dont_gc_arguments/must_gc_arguments - but I could change those too so they match.

Reply:
This code is a legacy carried over from x86. It's not necessary to "match". Changing others would be unnecessary churn, and risk breakage.

There's an urge from some contributors: when I suggest doing something in an easy-to-understand and clean way, people want to change everything else to match. This urge can be resisted, and IMVHO should be in this case. Churn is, in itself, bad.

And this case, is special, I think, because `does_not_return` uses the "don't do" anti-pattern, where the `true` case was `does_not_return`.

Reply:
That's fine, I'll leave that part as it. Knowing the motivation for this is useful, as every project is different.

Reply:
Sure, thanks. The commonality between this port and x86 means that in many cases we have taken x86 patches and applied them to this port, with a few tweaks. Of course the ports diverge over time, but even after almost ten years it still sometimes works.

Reply:
> 
> There's an urge from some contributors: when I suggest doing something in an easy-to-understand and clean way, people want to change everything else to match. This urge can be resisted, and IMVHO should be in this case. Churn is, in itself, bad.

Consistency is at least somewhat important though, right? This code is read much more often than it's modified, and in the case of the platform ports, often by people with less experience of OpenJDK as a whole. It seems worth spending a little time to do cleanups when modifying adjacent code, if it makes it easier to understand.

> 
> And this case, is special, I think, because `does_not_return` uses the "don't do" anti-pattern, where the `true` case was `does_not_return`.

`dont_gc_arguments` is equally or more confusing because its value is false and then it gets passed to an argument `must_gc_arguments` whose sense is inverted. I don't see what's wrong with:

```c++
StubFrame f(sasm, "blah", /* must_gc_arguments */ false, /* does_not_return */ true);
```

Reply:
> > There's an urge from some contributors: when I suggest doing something in an easy-to-understand and clean way, people want to change everything else to match. This urge can be resisted, and IMVHO should be in this case. Churn is, in itself, bad.
> 
> Consistency is at least somewhat important though, right?

Certainly, yes. But please resist the temptation to do things in a less-good way to be consistent with existing code. Code that has been running solidly for a decade is a benefit too. It's a judgement call. Hard-and-fast rules are a Bad Thing, because they get in the way of carefully-considered judgement calls.

> This code is read much more often than it's modified, and in the case of the platform ports, often by people with less experience of OpenJDK as a whole. It seems worth spending a little time to do cleanups when modifying adjacent code, if it makes it easier to understand.

True, but bear in mind that people also read diffs; and sometimes you have to spend time trying to figure out if a diff was actually meant to change something.

> `dont_gc_arguments` is equally or more confusing because its value is false and then it gets passed to an argument `must_gc_arguments` whose sense is inverted. I don't see what's wrong with:
> 
> ```c++
> StubFrame f(sasm, "blah", /* must_gc_arguments */ false, /* does_not_return */ true);
> ```

The fact that you need comments to mark the arguments is a code smell too, IMO. Sure, if you're going to pass a bunch of booleans the comments are essential. But we are where we are.

Reply:
Bear in mind also that this patch adds some complexity but provides (almost) no benefit to users. It's pretty marginal as a change, and therefore it's hard to justify much churn.